lib/cli/src/commands/create_exe.rs

```diff
@@ -419,32 +418,102 @@ impl CreateExe {
-                            files.into_iter().find(|f| f.contains(libwasmer_path)).ok_or_else(|| {
-                                anyhow!("Could not find libwasmer for {} target in the fetched release from Github: you can download it manually and specify its path with the --cross-compilation-library-path LIBRARY_PATH flag.", target)})?
+                            let _ = std::fs::create_dir_all(&target_file_path);
+                            let files = untar(local_tarball.clone(), target_file_path.clone())?;
+                            tarball_dir =
+                                target_file_path.canonicalize().unwrap_or(target_file_path);
+                            files.iter().find(|f| f.ends_with("libwasmer.a")).cloned().ok_or_else(|| {
+                                anyhow!("Could not find libwasmer.a for {} target in the provided tarball path (files = {files:#?})", target)})?
+                        } else {
+                            #[cfg(feature = "http")]
+                            {
+                                let release = http_fetch::get_latest_release()?;
+                                let tarball =
+                                    http_fetch::download_release(release, target.clone())?;
+                                let target_file_path = tarball
+                                    .parent()
+                                    .and_then(|parent| Some(parent.join(tarball.file_stem()?)))
+                                    .unwrap_or_else(|| tarball.clone());
+
+                                let target_file_path = target_file_path
+                                    .parent()
+                                    .and_then(|parent| {
+                                        Some(parent.join(target_file_path.file_stem()?))
+                                    })
+                                    .unwrap_or_else(|| target_file_path.clone());
+
+                                tarball_dir = target_file_path
+                                    .canonicalize()
+                                    .unwrap_or_else(|_| target_file_path.clone());
+                                let files = untar(tarball, target_file_path)?;
+                                files.iter().find(|f| f.ends_with("libwasmer.a")).cloned().ok_or_else(|| {
+                                    anyhow!("Could not find libwasmer for {} target in the fetched release from Github: you can download it manually and specify its path with the --cross-compilation-library-path LIBRARY_PATH flag. (files = {files:#?}", target)})?
+                            }
+                            #[cfg(not(feature = "http"))]
+                            return Err(anyhow!("This wasmer binary isn't compiled with an HTTP request library (feature flag `http`). To cross-compile, specify the path of the non-native libwasmer or release tarball with the --library-path LIBRARY_PATH or --tarball TARBALL_PATH flag."));
```

Comment:
Can we remove the `http` feature altogether and just make it accessible all the time? 

A `wasmer` CLI without internet access seems kinda pointless. Plus `wasmer create-exe` will always depend on resources we've downloaded from the internet, won't it?

Also, I suspect `reqwest` and `tokio` will already sneak in via our other dependencies (e.g. `wasmer-registry`), so the conditional compilation won't be helping us with build times.

Reply:
ok done